src/main/java/me/itzg/helpers/http/FilenameExtractor.java

```diff
@@ -24,69 +24,71 @@ public class FilenameExtractor {
 
-  private final LatchingUrisInterceptor interceptor;
+    private final LatchingUrisInterceptor interceptor;
 
-  public FilenameExtractor(LatchingUrisInterceptor interceptor) {
-    this.interceptor = Objects.requireNonNull(interceptor, "interceptor is required");
-  }
-
-  static String filenameFromContentDisposition(String headerValue) {
-      if (headerValue == null) {
-          return null;
-      }
+    public FilenameExtractor(LatchingUrisInterceptor interceptor) {
+        this.interceptor = Objects.requireNonNull(interceptor, "interceptor is required");
+    }
 
-      final ContentType parsed = ContentType.parse(headerValue);
-      log.debug("Response has contentDisposition={}", headerValue);
-      final String filenameStar = parsed.getParameter("filename*");
-      final String filename = parsed.getParameter("filename");
-      if (filenameStar != null) {
-          final Matcher m = RFC_5987_ENCODED.matcher(filenameStar);
-          if (m.matches()) {
-              try {
-                  return Rfc5987Util.decode(m.group(2), m.group(1));
-              } catch (UnsupportedEncodingException e) {
-                  throw new GenericException("Failed to decode filename* from " + headerValue, e);
-              }
-          }
-          else {
-              return filenameStar;
-          }
-      }
-      else if (filename != null) {
-          final Matcher m = RFC_2047_ENCODED.matcher(filename);
-          if (m.matches()) {
-              return m.group(1);
-          }
-          else {
-              return filename;
-          }
-      }
-      else {
-          throw new GenericException("Unable to determine filename from header: " + headerValue);
-      }
-  }
+    static @Nullable String filenameFromContentDisposition(String headerValue) {
+        if (headerValue == null) {
+            return null;
+        }
 
-  public String extract(ClassicHttpResponse response) throws IOException, ProtocolException {
-    // Same as AbstractHttpClientResponseHandler
-    if (response.getCode() >= HttpStatus.SC_REDIRECTION) {
-      EntityUtils.consume(response.getEntity());
-      throw new HttpResponseException(response.getCode(), response.getReasonPhrase());
+        final ContentType parsed = ContentType.parse(headerValue);
+        log.debug("Response has contentDisposition={}", headerValue);
+        final String filenameStar = parsed.getParameter("filename*");
+        final String filename = parsed.getParameter("filename");
+        if (filenameStar != null) {
+            final Matcher m = RFC_5987_ENCODED.matcher(filenameStar);
+            if (m.matches()) {
+                try {
+                    return Rfc5987Util.decode(m.group(2), m.group(1));
+                } catch (UnsupportedEncodingException e) {
+                    log.debug("Failed to decode filename* from {}", headerValue);
+                    return null;
```

Comment:
I think this scenario should be fatal. The filename value was non-null and it's trying to do the further processing on it.

Reply:
Just changed my mind, but make it a warning log. Since it's beyond the user's control what the server provides, it does seem fair to be non-fatal and best effort continue from there.

Reply:
Fixed